usecases/schema/tenant.go

```diff
@@ -51,48 +51,15 @@ func (h *Handler) AddTenants(ctx context.Context,
 		return 0, err
 	}
 
-	info, err := h.multiTenancy(class)
-	if err != nil {
-		return 0, err
-	}
-
-	names := make([]string, len(validated))
-	for i, tenant := range validated {
-		names[i] = tenant.Name
-	}
-
-	// create transaction payload
-	var partitions map[string][]string
-	f := func(_ *models.Class, st *sharding.State) (err error) {
-		if st == nil {
-			return fmt.Errorf("sharding state %w", ErrNotFound)
-		}
-		partitions, err = st.GetPartitions(h.clusterState, names, int64(info.ReplicationFactor))
-		return err
-	}
-
-	if err := h.metaReader.Read(class, f); err != nil {
-		return 0, fmt.Errorf("get partitions from class %q: %w", class, err)
-	}
-
-	if len(partitions) != len(names) {
-		h.logger.WithField("action", "add_tenants").
-			WithField("#partitions", len(partitions)).
-			WithField("#requested", len(names)).
-			Tracef("number of partitions for class %q does not match number of requested tenants", class)
-	}
 	request := api.AddTenantsRequest{
-		Tenants: make([]*api.Tenant, 0, len(partitions)),
+		ClusterNodes: h.clusterState.Candidates(),
+		Tenants:      make([]*api.Tenant, 0, len(validated)),
 	}
-	for i, name := range names {
-		part, ok := partitions[name]
-		if ok {
-			request.Tenants = append(request.Tenants, &api.Tenant{
-				Name:   name,
-				Nodes:  part,
-				Status: schema.ActivityStatus(validated[i].ActivityStatus),
-			})
-		}
+	for i, tenant := range validated {
+		request.Tenants = append(request.Tenants, &api.Tenant{
+			Name:   tenant.Name,
+			Status: schema.ActivityStatus(validated[i].ActivityStatus),
```

Comment:
**thought:**  i don't think we can end up in situation were  `validated[i]`  != `tenant` 

if we can use the same `tenant.ActivityStatus ` if  field exists

Reply:
They can differ if we have multiple tenant with the same name, we do the filtering in `validateTenants` if my memory is correct.